internal/querynodev2/segments/metricsutil/observer.go

```diff
@@ -0,0 +1,331 @@
+		Total:            int64(o.total.Get()),
+		Duration:         o.duration.Get(),
+		WaitLoadTotal:    int64(o.waitLoadTotal.Get()),
+		WaitLoadDuration: o.waitLoadDuration.Get(),
+	}
+}
+
+// newPromObserver creates a new promMetrics.
+func newPromObserver(nodeID string, label SegmentLabel) promMetricsObserver {
+	return promMetricsObserver{
+		nodeID:                               nodeID,
+		label:                                label,
+		DiskCacheLoadTotal:                   metrics.QueryNodeDiskCacheLoadTotal.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup),
+		DiskCacheLoadDuration:                metrics.QueryNodeDiskCacheLoadDuration.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup),
+		DiskCacheLoadBytes:                   metrics.QueryNodeDiskCacheLoadBytes.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup),
+		DiskCacheEvictTotal:                  metrics.QueryNodeDiskCacheEvictTotal.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup),
+		DiskCacheEvictDuration:               metrics.QueryNodeDiskCacheEvictDuration.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup),
+		QuerySegmentAccessTotal:              metrics.QueryNodeSegmentAccessTotal.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup, metrics.QueryLabel),
+		QuerySegmentAccessDuration:           metrics.QueryNodeSegmentAccessDuration.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup, metrics.QueryLabel),
+		QuerySegmentAccessWaitCacheTotal:     metrics.QueryNodeSegmentAccessWaitCacheTotal.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup, metrics.QueryLabel),
+		QuerySegmentAccessWaitCacheDuration:  metrics.QueryNodeSegmentAccessWaitCacheDuration.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup, metrics.QueryLabel),
+		SearchSegmentAccessTotal:             metrics.QueryNodeSegmentAccessTotal.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup, metrics.SearchLabel),
+		SearchSegmentAccessDuration:          metrics.QueryNodeSegmentAccessDuration.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup, metrics.SearchLabel),
+		SearchSegmentAccessWaitCacheTotal:    metrics.QueryNodeSegmentAccessWaitCacheTotal.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup, metrics.SearchLabel),
+		SearchSegmentAccessWaitCacheDuration: metrics.QueryNodeSegmentAccessWaitCacheDuration.WithLabelValues(nodeID, label.DatabaseName, label.ResourceGroup, metrics.SearchLabel),
+	}
+}
+
+// promMetricsObserver is a observer for prometheus metrics.
+type promMetricsObserver struct {
+	nodeID string
+	label  SegmentLabel // never updates
+
+	DiskCacheLoadTotal                   prometheus.Counter
+	DiskCacheLoadDuration                prometheus.Counter
+	DiskCacheLoadBytes                   prometheus.Counter
+	DiskCacheEvictTotal                  prometheus.Counter
+	DiskCacheEvictDuration               prometheus.Counter
```

Comment:
Why not use CacheAccess instead of QuerySegmentAccess and SearchSegmentAccess?

Reply:
cache load metric is record the real cache load operation, but not cache access.

cache access metric can be got in segment access related metric.